ibis/mapd/operations.py

```diff
@@ -0,0 +1,644 @@
+from datetime import date, datetime
+from ibis.mapd.identifiers import quote_identifier
+from six import StringIO
+from ibis.impala import compiler as impala_compiler
+
+import ibis
+import ibis.common as com
+import ibis.util as util
+import ibis.expr.datatypes as dt
+import ibis.expr.rules as rlz
+import ibis.expr.types as ir
+import ibis.expr.operations as ops
+
+_mapd_unit_names = {
```

Comment:
[resolution](https://github.com/ibis-project/ibis/blob/master/ibis/expr/datatypes.py#L416) property might be used for this